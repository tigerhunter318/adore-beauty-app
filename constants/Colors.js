const tintColor = '#2f95dc'

export default {
  tintColor,
  tabIconDefault: '#ccc',
  tabIconSelected: '#000',
  tabBar: '#fefefe',
  errorBackground: 'red',
  errorText: '#fff',
  warningBackground: '#EAEB5E',
  warningText: '#666804',
  noticeBackground: tintColor,
  noticeText: '#fff',
  orange: '#D6775E',
  mudColor: 'rgba(255, 175, 84, 0.49)',
  darkGray: '#404040',
  mediumGray: '#979797',
  lightPeachColor: 'rgba(251, 236, 223, 0.56)',
  peachColor: '#FBECDF',
  white: '#FFFFFF',
  titleColor: '#313131',
  lightPink: '#fdf6f0'
}
