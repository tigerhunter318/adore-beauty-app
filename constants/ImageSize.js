export default {
  scale: 2,
  product: {
    thumbnail: {
      width: 140, // match website
      height: 140
    },
    medium: {
      width: 140, // use thumbnail size to reduce network cache generation
      height: 140
    },
    large: {
      width: 300, // match website
      height: 300
    }
  },
  promotion: {
    thumbnail: {
      width: 105, // ?fm=webp&w=230&h=230
      height: 105
    }
  },
  brand: {
    listing: {
      width: 110,
      height: 110
    }
  },
  article: {
    avatar: {
      width: 40,
      height: 40 // fm=webp&w=50&h=50, ?fm=webp&w=70&h=70
    },
    thumbnail: {
      // width: 150, //?fm=webp&w=308&h=308, ?fm=webp&w=255&h=255
      // width: 164 + 0,
      // height: 126 + 0
      width: 100,
      height: 100
    },
    content: {
      // width: 330 + 0, // ?fm=webp&w=350&q=75
      // height: 210 + 0
      width: 325,
      height: 325
    },
    card: {
      // width: 380 + 0, // ?fm=webp&w=520&h=520, /?fm=webp&w=650&h=650
      // height: 200 + 0
      width: 325,
      height: 325
    }
  }
}
