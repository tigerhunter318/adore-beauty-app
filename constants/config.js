export default {
  recentProductsLimit: 10
}
