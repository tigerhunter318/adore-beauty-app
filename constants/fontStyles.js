import { StyleSheet } from 'react-native'

export const fontStyles = StyleSheet.create({
  h1: {
    fontSize: 32,
    lineHeight: 32
  },
  h2: {
    fontSize: 28,
    lineHeight: 32
  },
  h3: {
    fontSize: 23,
    lineHeight: 24
  },
  h4: {
    fontSize: 18,
    lineHeight: 24
  },
  h5: {
    fontSize: 16,
    lineHeight: 24
  },
  p: {
    fontSize: 16,
    lineHeight: 24,
    letterSpacing: 0.1
  }
})
