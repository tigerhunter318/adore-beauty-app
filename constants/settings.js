const settings = {
  defaultScrollEventThrottle: 8, // ios only https://reactnative.dev/docs/scrollview#scrolleventthrottle-ios,
  viewConfigRef: {
    waitForInteraction: false,
    itemVisiblePercentThreshold: 1
  }
}

export default settings
