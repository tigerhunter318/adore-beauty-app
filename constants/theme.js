import { Platform, PlatformColor } from 'react-native'

const theme = {
  black: '#000000',
  black10: 'rgba(0,0,0,0.1)',
  black20: 'rgba(0,0,0,0.2)',
  black30: 'rgba(0,0,0,0.3)',
  black40: 'rgba(0,0,0,0.4)',
  black50: 'rgba(0,0,0,0.5)',
  black60: 'rgba(0,0,0,0.6)',
  black70: 'rgba(0,0,0,0.7)',
  black80: 'rgba(0,0,0,0.8)',
  black90: 'rgba(0,0,0,0.9)',
  black100: 'rgba(0,0,0,1)',
  white: '#FFFFFF',
  green: '#4f8a0f',
  iOSKeyboard: '#d1d3d9',
  textGrey: '#BCBCBC',
  textGreyDark: '#777777',
  textMediumGrey: '#999999',
  borderColorDark: '#ACACAC',
  borderColor: '#E1DFDF',
  reviewBorderColor: '#EBEBEB',
  lightYellow: '#FBECDF',
  backgroundLightGrey: 'rgba(221,221,221,0.2)',
  lightGrey: '#f5f3f3',
  backgroundLighterGrey: 'rgba(225,223,223,0.41)',
  lightBlue: '#82B5C9',
  lightBackgroundBlue: '#f0f3f4',
  peach: '#FFA79D',
  darkBlack: '#212B36',
  lightBlack: '#404040',
  lighterBlack: '#4A4A4A',
  darkGray: '#979797',
  darkGrayWithOpacity: 'rgba(151, 151, 151, 0.2)',
  tabColor: 'rgba(102, 102, 102, 0.1)',
  pinkColor: '#E93777',
  titleBorderColor: '#201f26',
  red: '#FF0000',
  darkRed: '#cc0000',
  lighterRed: '#FFE0DC',
  lightPink: '#FBECDF',
  pink: '#FFCEC8',
  lighterPeach: '#fcf6ef',
  lighterPink: 'rgba(251, 236, 223, 0.5)',
  splitorColor: '#030303',
  orange: '#D6775E',
  orangeWithOpacity: 'rgba(214, 119, 94, 0.7)',
  darkerGray: 'rgba(36, 36, 36, 0.07)',
  lightBorder: '#CCCCCC',
  orangeColor: '#FFAF54',
  lightOrange: '#FECF96',
  codeOrange: '#FBECDE',
  lightPeach: '#FFCACB',
  backgroundGrey: 'rgba(242,242,242,1)',
  tabTextColor: 'rgba(74, 74, 74, 0.5)',
  darkOrange: '#F7C890',
  darkPeach: '#E3BDB1',
  tabBackgroundColor: 'rgba(216, 216, 216, 0.22)',
  level1: '#DB846D',
  level2: '#9DA483',
  level3: '#666666',
  giftIconLevel1: '#D6775E',
  giftIconLevel2: '#8BAC6C',
  giftIconLevel3: '#000000',
  stars: '#f7c868',
  linkColor: Platform.select({
    ios: PlatformColor('link'),
    android: PlatformColor('@android:color/holo_blue_light'),
    default: '#147EFB'
  })
}

export default theme
