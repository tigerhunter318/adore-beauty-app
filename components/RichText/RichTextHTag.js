import React from 'react'
import { findNodeLinkDetails } from './utils/helpers'
import Type from '../ui/Type'
import { fontStyles } from '../../constants/fontStyles'
import RichTextATag from './RichTextATag'
import { isValidArray } from '../../utils/validation'

const RichTextHTagNode = ({ node, onLinkPress, onProductPress }) => {
  const { name, parent, attribs, type, data, children } = node || {}
  const italic = !!(
    parent?.attribs?.style?.includes('italic') ||
    attribs?.style?.includes('italic') ||
    parent?.name === 'em' ||
    name === 'em'
  )
  const bold = !!(
    parent?.attribs?.style?.includes('bold') ||
    attribs?.style?.includes('bold') ||
    parent?.name === 'strong' ||
    name === 'strong'
  )
  const underline = !!(
    parent?.attribs?.style?.includes('underline') ||
    attribs?.style?.includes('underline') ||
    attribs?.href
  )

  if (type === 'text' && data) {
    return (
      <Type semiBold italic={italic} bold={bold} underline={underline}>
        {data}
      </Type>
    )
  }

  if (type === 'tag' && children?.length > 0) {
    if (name === 'a') {
      const linkDetails = findNodeLinkDetails(node)
      const handleLinkPress = () => onLinkPress(linkDetails.linkUrl)

      if (linkDetails.hasLink && linkDetails.linkTitle && typeof onLinkPress === 'function') {
        return (
          <RichTextATag
            node={node}
            onRedirect={handleLinkPress}
            semiBold
            onProductPress={onProductPress}
            {...linkDetails}
          />
        )
      }
    }

    return (
      <RichTextHTagNode
        italic={italic}
        bold={bold}
        underline={underline}
        node={{ name, parent, attribs, type, data, ...children[0] }}
      />
    )
  }
  return null
}

const RichTextHTag = ({ node, name, isNested = false, styleProps, onLinkPress, onProductPress }) => {
  const { hasLink } = findNodeLinkDetails(node)
  const centerText = !hasLink && (name === 'h3' || name === 'h4' || name === 'h5')

  return (
    <Type
      center={centerText}
      style={[fontStyles[name], styleProps?.[name]?.style]}
      mt={isNested ? 0 : 2}
      mb={isNested ? 0 : 2}
      {...(styleProps?.[name] || {})}
      semiBold
    >
      {isValidArray(node?.children) &&
        node.children.map((textNode, index) => (
          <RichTextHTagNode
            node={textNode}
            key={`${textNode.type}-${index}`}
            onLinkPress={onLinkPress}
            onProductPress={onProductPress}
          />
        ))}
    </Type>
  )
}

export default RichTextHTag
