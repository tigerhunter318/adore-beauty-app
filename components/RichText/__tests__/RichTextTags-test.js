import React from 'react'
import renderer from 'react-test-renderer'
import HTMLView from 'react-native-htmlview'
import { fontStyles } from '../../../constants/fontStyles'
import RichTextHTag from '../RichTextHTag'

const renderHtmlView = ({ html, isNested = true, styleProps }) => {
  const onLinkPress = jest.fn()
  return (
    <HTMLView
      value={html}
      addLineBreaks={false}
      renderNode={(node, index) => (
        <RichTextHTag
          node={node}
          key={`htag-${index}`}
          isNested={isNested}
          styleProps={styleProps}
          onLinkPress={onLinkPress}
        />
      )}
    />
  )
}

it(`RichTextHTag renders H1 tag`, () => {
  const html = '<h1>lorem ipsum h1</h1>'
  const tree = renderer.create(renderHtmlView({ html, styleProps: fontStyles.h1 })).toJSON()
  expect(tree).toMatchSnapshot()
})

it(`RichTextHTag renders H2 tag`, () => {
  const html = '<h1>lorem ipsum h2</h1>'
  const tree = renderer.create(renderHtmlView({ html, styleProps: fontStyles.h2 })).toJSON()
  expect(tree).toMatchSnapshot()
})

it(`RichTextHTag renders H3 tag`, () => {
  const html = '<h1>lorem ipsum h3</h1>'
  const tree = renderer.create(renderHtmlView({ html, styleProps: fontStyles.h3 })).toJSON()
  expect(tree).toMatchSnapshot()
})

it(`RichTextHTag renders H4 tag`, () => {
  const html = '<h1>lorem ipsum h4</h1>'
  const tree = renderer.create(renderHtmlView({ html, styleProps: fontStyles.h4 })).toJSON()
  expect(tree).toMatchSnapshot()
})

it(`RichTextHTag renders H5 tag`, () => {
  const html = '<h1>lorem ipsum h5</h1>'
  const tree = renderer.create(renderHtmlView({ html, styleProps: fontStyles.h5 })).toJSON()
  expect(tree).toMatchSnapshot()
})

it(`RichTextHTag renders H6 tag`, () => {
  const html = '<h1>lorem ipsum h6</h1>'
  const tree = renderer.create(renderHtmlView({ html, styleProps: fontStyles.h6 })).toJSON()
  expect(tree).toMatchSnapshot()
})

it(`RichTextHTag renders H tag with nested strong tag`, () => {
  const html = '<h1><strong>h1</strong> tag with nested strong tag</h1>'
  const tree = renderer.create(renderHtmlView({ html, styleProps: fontStyles.h6 })).toJSON()
  expect(tree).toMatchSnapshot()
})

it(`RichTextHTag renders H tag with nested strong and span tag`, () => {
  const html = '<h1><strong>h2</strong> tag with nested strong and <span>span</span> tag</h1>'
  const tree = renderer.create(renderHtmlView({ html, styleProps: fontStyles.h5 })).toJSON()
  expect(tree).toMatchSnapshot()
})
