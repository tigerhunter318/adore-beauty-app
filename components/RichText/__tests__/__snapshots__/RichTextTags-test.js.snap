// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RichTextHTag renders H tag with nested strong and span tag 1`] = `
<View>
  <Text
    allowFontScaling={false}
    mb={0}
    mt={0}
    style={
      Array [
        Object {
          "fontFamily": "Montserrat-SemiBold",
          "marginBottom": 0,
          "marginTop": 0,
        },
        Array [
          undefined,
          undefined,
        ],
      ]
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "fontFamily": "Montserrat-Bold",
          },
          undefined,
        ]
      }
    >
      h2
    </Text>
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "fontFamily": "Montserrat-SemiBold",
          },
          undefined,
        ]
      }
    >
       tag with nested strong and 
    </Text>
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "fontFamily": "Montserrat-SemiBold",
          },
          undefined,
        ]
      }
    >
      span
    </Text>
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "fontFamily": "Montserrat-SemiBold",
          },
          undefined,
        ]
      }
    >
       tag
    </Text>
  </Text>
</View>
`;

exports[`RichTextHTag renders H tag with nested strong tag 1`] = `
<View>
  <Text
    allowFontScaling={false}
    mb={0}
    mt={0}
    style={
      Array [
        Object {
          "fontFamily": "Montserrat-SemiBold",
          "marginBottom": 0,
          "marginTop": 0,
        },
        Array [
          undefined,
          undefined,
        ],
      ]
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "fontFamily": "Montserrat-Bold",
          },
          undefined,
        ]
      }
    >
      h1
    </Text>
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "fontFamily": "Montserrat-SemiBold",
          },
          undefined,
        ]
      }
    >
       tag with nested strong tag
    </Text>
  </Text>
</View>
`;

exports[`RichTextHTag renders H1 tag 1`] = `
<View>
  <Text
    allowFontScaling={false}
    mb={0}
    mt={0}
    style={
      Array [
        Object {
          "fontFamily": "Montserrat-SemiBold",
          "marginBottom": 0,
          "marginTop": 0,
        },
        Array [
          undefined,
          undefined,
        ],
      ]
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "fontFamily": "Montserrat-SemiBold",
          },
          undefined,
        ]
      }
    >
      lorem ipsum h1
    </Text>
  </Text>
</View>
`;

exports[`RichTextHTag renders H2 tag 1`] = `
<View>
  <Text
    allowFontScaling={false}
    mb={0}
    mt={0}
    style={
      Array [
        Object {
          "fontFamily": "Montserrat-SemiBold",
          "marginBottom": 0,
          "marginTop": 0,
        },
        Array [
          undefined,
          undefined,
        ],
      ]
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "fontFamily": "Montserrat-SemiBold",
          },
          undefined,
        ]
      }
    >
      lorem ipsum h2
    </Text>
  </Text>
</View>
`;

exports[`RichTextHTag renders H3 tag 1`] = `
<View>
  <Text
    allowFontScaling={false}
    mb={0}
    mt={0}
    style={
      Array [
        Object {
          "fontFamily": "Montserrat-SemiBold",
          "marginBottom": 0,
          "marginTop": 0,
        },
        Array [
          undefined,
          undefined,
        ],
      ]
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "fontFamily": "Montserrat-SemiBold",
          },
          undefined,
        ]
      }
    >
      lorem ipsum h3
    </Text>
  </Text>
</View>
`;

exports[`RichTextHTag renders H4 tag 1`] = `
<View>
  <Text
    allowFontScaling={false}
    mb={0}
    mt={0}
    style={
      Array [
        Object {
          "fontFamily": "Montserrat-SemiBold",
          "marginBottom": 0,
          "marginTop": 0,
        },
        Array [
          undefined,
          undefined,
        ],
      ]
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "fontFamily": "Montserrat-SemiBold",
          },
          undefined,
        ]
      }
    >
      lorem ipsum h4
    </Text>
  </Text>
</View>
`;

exports[`RichTextHTag renders H5 tag 1`] = `
<View>
  <Text
    allowFontScaling={false}
    mb={0}
    mt={0}
    style={
      Array [
        Object {
          "fontFamily": "Montserrat-SemiBold",
          "marginBottom": 0,
          "marginTop": 0,
        },
        Array [
          undefined,
          undefined,
        ],
      ]
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "fontFamily": "Montserrat-SemiBold",
          },
          undefined,
        ]
      }
    >
      lorem ipsum h5
    </Text>
  </Text>
</View>
`;

exports[`RichTextHTag renders H6 tag 1`] = `
<View>
  <Text
    allowFontScaling={false}
    mb={0}
    mt={0}
    style={
      Array [
        Object {
          "fontFamily": "Montserrat-SemiBold",
          "marginBottom": 0,
          "marginTop": 0,
        },
        Array [
          undefined,
          undefined,
        ],
      ]
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "fontFamily": "Montserrat-SemiBold",
          },
          undefined,
        ]
      }
    >
      lorem ipsum h6
    </Text>
  </Text>
</View>
`;
